runelite-client/src/main/java/net/runelite/client/plugins/timetracking/farming/FarmingContractManager.java

```diff
@@ -0,0 +1,351 @@
+		@Nonnull ConfigManager configManager
+	)
+	{
+		this.client = client;
+		this.itemManager = itemManager;
+		this.config = config;
+		this.plugin = plugin;
+		this.farmingWorld = farmingWorld;
+		this.farmingTracker = farmingTracker;
+		this.infoBoxManager = infoBoxManager;
+		this.configManager = configManager;
+	}
+
+	public void setContract(@Nullable FarmingContract contract)
+	{
+		this.contract = contract;
+		setStoredContract(contract);
+		handleContractState();
+	}
+
+	public boolean hasContract()
+	{
+		return contract != null;
+	}
+
+	@Nullable
+	public Tab getContractTab()
+	{
+		return hasContract() ? contract.getPatchImplementation().getTab() : null;
+	}
+
+	@Nullable
+	public String getContractName()
+	{
+		return hasContract() ? contract.getName() : null;
+	}
+
+	public boolean shouldHighlightFarmingTabPanel(@Nonnull FarmingPatch patch)
+	{
+		PatchPrediction patchPrediction = farmingTracker.predictPatch(patch);
+		if (contract != null &&
+			patch.getRegion().getName().equalsIgnoreCase(FARMING_REGION_NAME) &&
+			contract.getPatchImplementation() == patch.getImplementation() &&
+			patchPrediction != null &&
+			(patchPrediction.getProduce().equals(contract.getProduce()) || summary == SummaryState.EMPTY))
+		{
+			return true;
+		}
+		return false;
+	}
+
+	public void loadContractFromConfig()
+	{
+		contract = getStoredContract();
+		handleContractState();
+	}
+
+	public boolean updateData(WorldPoint loc)
+	{
+		SummaryState oldSummary = summary;
+
+		handleContractState();
+		if (loc.getRegionID() == FARMING_GUILD_REGION_ID)
+		{
+			handleGuildmasterJaneWidgetDialog();
+			handleInfoBox();
+		}
+		else
+		{
+			if (infoBox != null)
+			{
+				infoBoxManager.removeInfoBox(infoBox);
+				infoBox = null;
+			}
+		}
+		return oldSummary != summary;
+	}
+
+	private void handleInfoBox()
+	{
+		if (infoBox != null)
+		{
+			if (contract == null)
+			{
+				infoBoxManager.removeInfoBox(infoBox);
+				infoBox = null;
+			}
+			else if (infoBox.getContract() != contract)
+			{
+				infoBoxManager.removeInfoBox(infoBox);
+				infoBox = new FarmingContractInfoBox(itemManager.getImage(contract.getProduce().getItemID()), plugin, contract, config, this);
+				infoBoxManager.addInfoBox(infoBox);
+			}
+		}
+		else if (contract != null)
+		{
+			infoBox = new FarmingContractInfoBox(itemManager.getImage(contract.getProduce().getItemID()), plugin, contract, config, this);
+			infoBoxManager.addInfoBox(infoBox);
+		}
+	}
+
+	private void handleGuildmasterJaneWidgetDialog()
+	{
+		Widget npcDialog = client.getWidget(WidgetInfo.DIALOG_NPC_HEAD_MODEL);
+
+		if (npcDialog == null || npcDialog.getModelId() != GUILDMASTER_JANE_WIDGET_MODEL_ID)
+		{
+			return;
+		}
+
+		String dialogText = Text.removeTags(client.getWidget(WidgetInfo.DIALOG_NPC_TEXT).getText());
+		log.debug(dialogText + " ");
+
+		if (dialogText.equals(CONTRACT_REWARDED))
+		{
+			setContract(null);
+			;
+		}
+		handleContractAssigning(dialogText);
```

Comment:
just remove these 4 lines, there is no reason for this to be its own method when you are just tail calling it